ambry-replication/src/main/java/com/github/ambry/replication/ReplicationManager.java

```diff
@@ -331,6 +333,69 @@ public void onReplicaAddedOrRemoved(List<ReplicaId> addedReplicas, List<ReplicaI
+      // 1. We are only updating the existing partitions (not adding or removing) in the 'peerLeaderReplicasByPartition' map. Since, it is a concurrent hash map, PUTs and GETs are clean.
+      // 2. Updating of Replica state (going to be maintained in RemoteReplicaInfo class) of newly found remote leaders and old leaders will be synchronized in RemoteReplicaInfo class.
+
+      rwlock1.readLock().lock();
+      try {
+        for (String partitionName : peerLeaderReplicasByPartition.keySet()) {
+          ReplicaId localLeaderReplica = storeManager.getReplica(partitionName);
+          PartitionId partition = localLeaderReplica.getPartitionId();
+          List<ReplicaId> currentRemoteLeaderReplicas = peerLeaderReplicasByPartition.get(partitionName);
+          List<ReplicaId> updatedRemoteLeaderReplicas = partition.getReplicaIdsByState(ReplicaState.LEADER, null)
+              .stream()
+              .filter(r -> !r.getDataNodeId().getDatacenterName().equals(dataNodeId.getDatacenterName()))
+              .collect(Collectors.toList());
+
+          //Collect the set of new remote leader replicas
+          Set<ReplicaId> addedRemoteReplicas = new HashSet<>(updatedRemoteLeaderReplicas);
+          addedRemoteReplicas.removeAll(currentRemoteLeaderReplicas);
+
+          //Collect the set of old remote leader replicas
+          Set<ReplicaId> removedRemoteReplicas = new HashSet<>(currentRemoteLeaderReplicas);
+          removedRemoteReplicas.removeAll(updatedRemoteLeaderReplicas);
+
+          for (ReplicaId remoteReplica : addedRemoteReplicas) {
+            // for now, we are just logging the newly found remote leader replicas
+            // we will update the Replica state (going to be maintained in RemoteReplicaInfo object) of the relevant remote replicas in later PRs
+            logger.info("Adding new remote leader {} for Partition {} to replicate from dc {}",
+                getInstanceName(remoteReplica.getDataNodeId().getHostname(), remoteReplica.getDataNodeId().getPort()),
+                partitionName, remoteReplica.getDataNodeId().getDatacenterName());
+          }
+
+          //remove replicaInfo from existing partitionInfo and replica-threads
+          List<RemoteReplicaInfo> replicaInfosToRemove = new ArrayList<>();
+          for (ReplicaId remoteReplica : removedRemoteReplicas) {
+            // for now, we are just logging the old leader replicas
+            // we will update the Replica state (going to be maintained in RemoteReplicaInfo object) of the relevant remote replicas in later PRs
+            logger.info("Removing old remote leader {} for Partition {} from dc {}",
+                getInstanceName(remoteReplica.getDataNodeId().getHostname(), remoteReplica.getDataNodeId().getPort()),
+                partitionName, remoteReplica.getDataNodeId().getDatacenterName());
+          }
+
+          peerLeaderReplicasByPartition.put(partitionName, updatedRemoteLeaderReplicas);
```

Comment:
I know in current PR, `peerLeaderReplicasByPartition` is only updated but not really used. In future PR, will it be passed into `ReplicaThread`?